lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1002,7 +1003,66 @@ struct LoadOpConversion
       const size_t nWords = std::max<size_t>(1, totalWidth / width);
       const size_t wordNElems = width / valueElemNbits;
       assert(wordNElems * nWords * numVecs == numElems);
+#ifdef USE_ROCM
+      GCNBuilder gcnBuilder;
+
+      const std::string readConstraint = "v";
+      const std::string writeConstraint = "=&v";
+
+      auto *dstsOpr = gcnBuilder.newListOperand();
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        auto *opr = gcnBuilder.newOperand(writeConstraint); // =v operations
+        dstsOpr->listAppend(opr);
+      }
+
+      auto *addrOpr = gcnBuilder.newAddrOperand(ptrElems[vecStart], "v");
+      auto *offOpr = gcnBuilder.newEmptyOperand("off");
+
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        auto &gload = gcnBuilder.create<GCNMemInstr>("global_load")
+                          ->load_type(valueElemNbits);
+        unsigned offset = wordIdx * (valueElemNbits / 8);
+        auto *offsetMod =
+            gcnBuilder.newModifier("offset", std::to_string(offset));
+        gload({dstsOpr->listGet(wordIdx), addrOpr, offOpr}, {offsetMod});
+      }
+
+      auto &wait_cnt = *gcnBuilder.create<>("s_waitcnt vmcnt(0)");
+      wait_cnt();
```

Comment:
@B1tway This hurts performance badly. Can it be removed and let the compiler decides when `s_waitcnt` be inserted?